cylc/cylc-ui
# tree: more efficient expand/collapse buttons

* Closes https://github.com/cylc/cylc-ui/issues/1617
* There is one expand/collapse button per tree item.
* There are many tree items in the tree view.
* We were using two Vuetify components to implement the expand/collapse button, these components are expensive to mount.
* This PR switches to using native components to implement the expand/collapse button.
* This saves ~2/5 of mount+render time.

To test:
* Start the workflow from https://github.com/cylc/cylc-ui/issues/1614 with `hours` set to `20` in paused mode.
* Open the tree view on this workflow and wait for the UI to stabilise.
* Time how long it takes to open a second tree view.

Results:
* Before: ~5s
* After: ~2.5s

Caveats:
* This approach does not support standard Vuetify things like the mouseover "ripple". Personally I see this as more of a pro than a con but it's a change either way.
* This approach does not re-implement the rotate animation on state change. This would be fairly easy to do with CSS, personally I'm happier without, but it's another change.

**Check List**
